config/v1/types/filesystem_test.go

```diff
@@ -77,118 +43,14 @@ func TestFilesystemFormatAssertValid(t *testing.T) {
 		},
 		{
 			in:  in{format: FilesystemFormat("")},
-			out: out{err: ErrFilesystemInvalidFormat},
-		},
-	}
-
-	for i, test := range tests {
-		err := test.in.format.AssertValid()
-		if !reflect.DeepEqual(test.out.err, err) {
-			t.Errorf("#%d: bad error: want %v, got %v", i, test.out.err, err)
-		}
-	}
-}
-
-func TestMkfsOptionsUnmarshalJSON(t *testing.T) {
-	type in struct {
-		data string
-	}
-	type out struct {
-		options MkfsOptions
-		err     error
-	}
-
-	tests := []struct {
-		in  in
-		out out
-	}{
-		{
-			in:  in{data: `["--label=ROOT"]`},
-			out: out{options: MkfsOptions([]string{"--label=ROOT"})},
-		},
-	}
-
-	for i, test := range tests {
-		var options MkfsOptions
-		err := json.Unmarshal([]byte(test.in.data), &options)
-		if !reflect.DeepEqual(test.out.err, err) {
-			t.Errorf("#%d: bad error: want %v, got %v", i, test.out.err, err)
-		}
-		if !reflect.DeepEqual(test.out.options, options) {
-			t.Errorf("#%d: bad format: want %#v, got %#v", i, test.out.options, options)
-		}
-	}
-}
-
-func TestFilesystemUnmarshalJSON(t *testing.T) {
-	type in struct {
-		data string
-	}
-	type out struct {
-		filesystem Filesystem
-		err        error
-	}
-
-	tests := []struct {
-		in  in
-		out out
-	}{
-		{
-			in:  in{data: `{"device": "/foo", "format": "ext4"}`},
-			out: out{filesystem: Filesystem{Device: "/foo", Format: "ext4"}},
-		},
-		{
-			in:  in{data: `{"format": "ext4"}`},
-			out: out{filesystem: Filesystem{Format: "ext4"}, err: ErrPathRelative},
-		},
-	}
-
-	for i, test := range tests {
-		var filesystem Filesystem
-		err := json.Unmarshal([]byte(test.in.data), &filesystem)
-		if !reflect.DeepEqual(test.out.err, err) {
-			t.Errorf("#%d: bad error: want %v, got %v", i, test.out.err, err)
-		}
-		if !reflect.DeepEqual(test.out.filesystem, filesystem) {
-			t.Errorf("#%d: bad filesystem: want %#v, got %#v", i, test.out.filesystem, filesystem)
-		}
-	}
-}
-
-func TestFilesystemAssertValid(t *testing.T) {
-	type in struct {
-		filesystem Filesystem
-	}
-	type out struct {
-		err error
-	}
-
-	tests := []struct {
-		in  in
-		out out
-	}{
-		{
-			in:  in{filesystem: Filesystem{Device: "/foo", Format: "ext4"}},
```

Comment:
Looks like we're losing these tests.

Reply:
There is no longer an `AssertValid` for `Filesystem` since it just called `AssertValid` on two of its fields of types `FilesystemFormat` and `Path`. The logic that finds and calls the `Validate` functions is what guarantees that the sub-`Validate`s are called, and there are separate tests for `Validate` on both of those fields ([here](https://github.com/coreos/ignition/pull/519/files/530602f5a9eae4b25e85a425befc49ec99b02cba..c667b7faec249bfe32db53595a4b843cfd26a66f#diff-b5055fb9d5075288ed7b1b69531a5e2fR24) and [here](https://github.com/coreos/ignition/pull/519/files/530602f5a9eae4b25e85a425befc49ec99b02cba..c667b7faec249bfe32db53595a4b843cfd26a66f#diff-adfe4224d4d2d5a075e2183a18298cdaR24)).